test/cluster-tests/kubernetes/test.sh

```diff
@@ -94,13 +99,21 @@ fi
 wait=5
 count=12
 while (( $count > 0 )); do
-  running=$(kubectl get pods --namespace=kube-system | grep kube-proxy | grep Running | wc | awk '{print $1}')
-  if (( ${running} == ${EXPECTED_NODE_COUNT} )); then break; fi
+  nonrunning=$(kubectl get pods --namespace=kube-system | grep kube-proxy | grep -v Running | wc | awk '{print $1}')
+  if (( ${nonrunning} == 0 )); then break; fi
   sleep 5; count=$((count-1))
 done
-if (( ${running} != ${EXPECTED_NODE_COUNT} )); then
-  echo "gave up waiting for kube-proxies"; exit -1
-fi
+
+# get master public hostname
+master=$(kubectl config view | grep server | cut -f 3- -d "/" | tr -d " ")
```

Comment:
You can pass output specifiers to `kubectl`. May or may not be less elegant than grep/cut/tr. Just a heads up more than anything else.

For example, this is how I retrieve the service LoadBalancer IP:  `kubectl get svc --namespace polykube nginx-ingress --template="{{range .status.loadBalancer.ingress}}{{.ip}}{{end}}"`